test/utils/fees.ts

```diff
@@ -1,27 +1,25 @@
-import {calldataCost} from "@eth-optimism/core-utils";
-import {formatEther} from "viem";
-import {ETH_PRICE_USD, L1_GAS_PRICE} from "../../hardhat.config";
+import {Hex, formatEther, hexToBytes} from "viem";
+import {ETH_PRICE_USD} from "../../hardhat.config";
 
-const OP_FIXED_OVERHEAD = 188;
-const OP_DYNAMIC_OVERHEAD_SCALAR = 0.684;
-const SCALAR_DECIMALS = 3;
+// from https://github.com/ethereum-optimism/optimism/blob/ba174f4d5f4020ff16298fefd86b55a29d4724a9/packages/contracts-bedrock/src/L2/GasPriceOracle.sol#L24
+const DECIMALS = 6n;
+const L1_BASE_FEE = BigInt(process.env.L1_BASE_FEE || "");
+const L1_BLOB_BASE_FEE = BigInt(process.env.L1_BLOB_BASE_FEE || "");
+const L1_BASE_FEE_SCALAR = BigInt(process.env.L1_BASE_FEE_SCALAR || "");
+const L1_BLOB_BASE_FEE_SCALAR = BigInt(
+  process.env.L1_BLOB_BASE_FEE_SCALAR || "",
+);
 
-/**
- * NOTE: The Optimism docs also factor in the dynamic overhead into the gas
- * used, but Etherscan does not. We follow Etherscan here and multiply by the
- * dynamic overhead during the fee calculation.
- */
-export function getL1GasUsed(serializedTx: `0x${string}`) {
-  return calldataCost(serializedTx).toBigInt() + BigInt(OP_FIXED_OVERHEAD);
+export function getL1GasUsed(serializedTx: Hex) {
+  return calldataGas(serializedTx);
```

Comment:
Can you revert the uninstall of `@eth-optimism/core-utils` and use `calldataCost` instead? The current implementation is inaccurate because of `+ 68n * 16n` in line 37.